File: openmp/exercises/exercise_2/solution.c
#include <stdio.h>
#include <omp.h>

#define ARRAY_SIZE 10

int main(int argc, char* argv[])
{
	int a[ARRAY_SIZE];

	// 1) Create the OpenMP parallel region
	#pragma omp parallel
	{
		// 1.1) Create the for construct and initialise the array elements
		#pragma omp for
		for(int i = 0; i < ARRAY_SIZE; i++)
		{
			a[i] = i * i;
		}
	}

	// 2) Print the array elements, sequentially
	for(int i = 0; i < ARRAY_SIZE; i++)
	{
		printf("a[%d] = %2d\n", i, a[i]);
	}

	return 0;
}

File: openmp/exercises/exercise_2/solution.f90
PROGRAM main
    IMPLICIT NONE

    INTEGER, PARAMETER :: ARRAY_SIZE = 10
    INTEGER :: a(0:ARRAY_SIZE-1)
    INTEGER :: i

    ! 1) Create the OpenMP parallel region
    !$OMP parallel default(none) shared(a,i)
        ! 1.1) Create the for construct and initialise the array elements
        !$OMP do
        DO i = 0, ARRAY_SIZE-1
            a(i) = i * i
        END DO
    !$OMP END parallel

    ! 2) Print the array elements, sequentially
    DO i = 0, ARRAY_SIZE-1
        WRITE(*, '(A,I0,A,I2)') 'a(', i, ') = ', a(i)
    END DO
END PROGRAM main